drivers/sensor/ti/ina7xx/ina7xx.c

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright The Zephyr Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_ina7xx
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include "ina7xx.h"
+
+LOG_MODULE_REGISTER(INA7xx, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MAX_RETRIES 4
+
+struct ina7xx_config {
+	struct i2c_dt_spec bus;
+	uint8_t inatype;
+	uint8_t mode;
+	uint8_t convdly;
+	uint8_t vbusct;
+	uint8_t vsenct;
+	uint8_t tct;
+	uint8_t avg;
+};
+
+struct ina7xx_data {
+	uint16_t config;
+	uint8_t convdly;
+	uint16_t adc_config;
+	uint32_t v_bus;
+	uint16_t die_temp;
+	uint16_t current;
+	uint64_t power;
+	uint8_t valid_readings_mask;
+	uint16_t current_mul_ua;
+	uint16_t power_mul_uw;
+};
+
+static int ina7xx_read(const struct device *dev, uint8_t reg_addr, void *reg_data, uint8_t bytes)
+{
+	const struct ina7xx_config *cfg = dev->config;
+	uint8_t rx_buf[bytes];
+	int rc;
+
+	rc = i2c_write_read_dt(&cfg->bus, &reg_addr, sizeof(reg_addr), rx_buf, sizeof(rx_buf));
+
+	if (bytes == 2) {
+		*(uint16_t *)reg_data = sys_get_be16(rx_buf);
+	} else if (bytes == 3) {
+		*(uint64_t *)reg_data = sys_get_be24(rx_buf);
+	} else if (bytes == 4) {
+		*(uint64_t *)reg_data = sys_get_be32(rx_buf);
+	} else {
+		*(uint64_t *)reg_data = sys_get_be40(rx_buf);
+	}
+
+	return rc;
+}
+
+static int ina7xx_write(const struct device *dev, uint8_t addr, uint16_t reg_data)
+{
+	const struct ina7xx_config *cfg = dev->config;
+	uint8_t tx_buf[3];
+
+	tx_buf[0] = addr;
+	sys_put_be16(reg_data, &tx_buf[1]);
+
+	return i2c_write_dt(&cfg->bus, tx_buf, sizeof(tx_buf));
+}
+
+static int ina7xx_reg_field_update(const struct device *dev, uint8_t addr, uint16_t mask,
+				   uint16_t field)
+{
+	uint16_t reg_data;
+	int rc;
+
+	rc = ina7xx_read(dev, addr, &reg_data, 2);
+	if (rc) {
```

Comment:
```suggestion
	if (rc != 0) {
```
A small one for here and other occurrences. This is Zephyr rule 85 https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html

Reply:
i changed it.